src/components/Posts/components/types.ts

```diff
@@ -0,0 +1,7 @@
+import { IPost } from '../types';
+
+export interface IPostComponent {
```

Comment:
```suggestion
export interface IPostComponentProps {
```
`Props` Postfix will help you to recognise props-types